### PR TITLE
treewide/applications: add `sourceType` `binaryNativeCode` for many packages

### PR DESCRIPTION
**Pull request by risicle**
###### Description of changes

Again, not comprehensive. These are packages where binary native code makes it from the downloaded `src` to the output.

Done for both free and unfree packages as it seemed easier not to differentiate when it came to it.

Skips AppImages as I'm guessing it'll be easier to handle them all together. (edit see #177308)

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
:confused: :interrobang: 
what changed about `mendeley`?

**Comment by SuperSandro2000** [comment]
Maybe because it is unfree?

**Comment by risicle** [comment]
There are plenty of other unfree packages I touched here though.

**Comment by risicle** [comment]
I'm not tremendously worried about it TBH..

**Comment by risicle** [comment]
@ofborg eval

File: pkgs/applications/audio/baudline/default.nix [hunk 1]
```diff
@@ -63,6 +63,7 @@ stdenv.mkDerivation rec {
     # See http://www.baudline.com/faq.html#licensing_terms.
```
Comment (kevincox) [inline pkgs/applications/audio/baudline/default.nix:63]: Can you share a bit of info how this change was generated? Particularly how were "packages where binary native code makes it from the downloaded src to the output" packages identified?

Comment (risicle) [reply pkgs/applications/audio/baudline/default.nix:63]: A few methods.

A first pass to find candidates used a grep for either architecture names in urls or uses of `patchelf`. Then I would look manually at the source package(s) if I had any doubts and try and figure out if the binaries made it to the output.

**Review by Mindavi — APPROVED** [review]

**Comment by Mindavi** [comment]
Cherry picked some tens of packages and checked, all looked ok.

**Pull request merged**